### PR TITLE
Build & deploy Docs automatically to gh-pages

### PR DESCRIPTION
**Pull request by Jesus89**
Related to: #1741

With each v4 commit, the API documentation is built with [JSDoc](http://usejsdoc.org/) and commited to [gh-pages branch](https://github.com/CartoDB/cartodb.js/commits/gh-pages). No generated docs in v4 branch!

This is the main page: http://cartodb.github.io/cartodb.js/

### CODE REVIEW

File: scripts/deploy-to-gh-pages.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+# Pull requests and commits to other branches shouldn't try to deploy
+if [ "$TRAVIS_PULL_REQUEST" != "false" ] && [ "$TRAVIS_BRANCH" != "jsdoc-ghpages" ]; then
```
Comment (alonsogarciapablo) [inline scripts/deploy-to-gh-pages.sh]: "jsdoc-ghpages"? It should say "v4", right?

File: .travis.yml [hunk 2]
```diff
@@ -1,19 +1,30 @@
...
+  - bash scripts/deploy-to-gh-pages.sh
 notifications:
   email:
     on_success: never
     on_failure: change
+env:
+  global:
+    secure: D8bXGaTynX3VVPQtwZC03KX95EeBDxzlqhDnq6XskNcWbKwC9PBCzb/c3PF4184ra+XRf5uZZfaQjlZTlGWGfD0xCZwI6TuX0Qj2XfQT/nWhrHrFYm7oL4JeunYSIj2tv0wkIcSJL1/fsc49o+Geqf+v96HGYG1s/Sq6ig60/wQ=
+
+#deploy:
```
Comment (alonsogarciapablo) [inline .travis.yml]: 🔥 ?

File: scripts/deploy-to-gh-pages.sh (hunk 1 above)
Comment (Jesus89) [reply scripts/deploy-to-gh-pages.sh]: Yes, before merging

File: .travis.yml (hunk 2 above)
Comment (Jesus89) [reply .travis.yml]: This is to (secure) push commits to gh-pages

**Comment by Jesus89** [comment]
In the @api tag PR I will include v4 docs for API docs and internal for all the docs

**Comment by Jesus89** [comment]
We can also use the @ignore tag for completely ignored comments.

File: .travis.yml (hunk 2 above)
Comment (alonsogarciapablo) [reply .travis.yml]: But it's commented out, right?

Comment (Jesus89) [reply .travis.yml]: Oh, the commented part is just a test. I will remove it.

**Comment by alonsogarciapablo** [comment]
> We can also use the @ignore tag for completely ignored comments.

Yes, I was thinking perhaps we could @ignore all the JSDocs that we have so far so that we can then unignore things that might make sense to have documented for real. We can also merge this and review existing docs in another PR. What do you think?

**Comment by Jesus89** [comment]
Yes, we can merge this. We will check doc and @api / @ignore tags in https://github.com/CartoDB/cartodb.js/issues/1764.

**Comment by alonsogarciapablo** [comment]
:+1:

**Comment by alonsogarciapablo** [comment]
cc: @csubira

**Pull request merged**